validate.go

```diff
@@ -225,6 +226,37 @@ func (v *Validator) Domain(key, value string, message ...string) {
 	}
 }
 
+// URL validates that the string contains a valid URL.  The scheme is optional.
+// The URL may consist of a scheme, host, path, and query parameters.  Only the
+// host is required.  If the scheme is not given it will be prepended.
+func (v *Validator) URL(key, value string, message ...string) {
+	if value == "" {
+		return
+	}
+
+	msg := getMessage(message, MessageURL)
+
+	u, err := url.Parse(value)
+	// If we don't have a scheme the parse may or may not fail according to the
+	// go docs. "Trying to parse a hostname and path without a scheme is invalid
+	// but may not necessarily return an error, due to parsing ambiguities."
+	if u.Scheme == "" {
+		u.Scheme = "http"
+		u, err = url.Parse(u.String())
+	}
+
+	if err != nil {
+		v.Append(key, "%s: %s", msg, err)
+		return
+	}
+
+	if u.Host == "" {
+		v.Append(key, "no host given: %s", u.String())
```

Comment:
All messages can be translated, so would be good to make sure this one can as well.

I think just using `MessageURL` would be fine?

Reply:
Though according to the spec we aren't supposed to translate i18n.

Reply:
Sure, but this isn't about the spec, this is a separate library which is quite consistent in its i18n support except for here.